modules/layers/src/scatterplot-layer/scatterplot-layer.ts

```diff
@@ -233,6 +236,7 @@ export default class ScatterplotLayer<DataT = any, ExtraPropsT extends {} = {}>
     super.updateState(params);
 
     if (params.changeFlags.extensionsChanged) {
+      this.shaderInputs = new ShaderInputs<{picking: typeof picking.props}>({picking} as any);
```

Comment:
Not familiar with this new class - Is it necessary to create a new one for every model instance?

Reply:
The [aim](https://github.com/visgl/deck.gl/pull/8334#issuecomment-1837153551) is to have layers share UBOs for things like projection & picking. This has no yet been implemented in luma, as [`Model.setShaderInputs()`](https://github.com/visgl/luma.gl/blob/0b1bfcabb89ad751ca4c27a44caee4c010be7ba6/modules/engine/src/model/model.ts#L371) is always creating a new UBO via `new UniformStore()`

As the shared UBOs will be closely coupled to the `ShaderModule` I'm thinking of adding a property e.g. `ShaderModule.uniformBuffer` which would then be used by the `UniformStore` rather than creating a new UBO for each `Model`. @ibgreen is this how you were imagining it?

Reply:
I am not sure a completely global module is appropriate. 
- At minimum one "global" UBO is needed per device. 
- And sometimes one wants to share material uniform buffers between nodes in a glTF scenegraph etc.

As discussed we'll drop UBO sharing for now, we can come back to it in v9.1 and focus on getting this landed. 

Since you are now using the default ShaderInputs created by the model the objection of @Pessimistress should be resolved for now as not much is visible in deck.gl.